cloudslang-content-maven-compiler/src/main/java/io/cloudslang/maven/compiler/CloudSlangMavenCompiler.java

```diff
@@ -0,0 +1,148 @@
+        return false;
+    }
+
+    public CompilerResult performCompile(CompilerConfiguration config) throws CompilerException {
+        CompilerResult compilerResult = new CompilerResult();
+        List<CompilerMessage> compilerMessage = new ArrayList();
+
+        //we do not want the source files that were calculated because we have multiple suffix
+        //and the framework support only one via the inputFileEnding
+        config.setSourceFiles(null);
+        String[] sourceFiles = getSourceFiles(config);
+        if(sourceFiles.length > 0) {
+            System.out.println("Compiling " + sourceFiles.length + " " + "source file" + (sourceFiles.length == 1?"":"s"));
+            for (String sourceFile : sourceFiles) {
+                compilerMessage.addAll(compileFile(sourceFile));
+            }
+
+            if (compilerMessage.size() > 0){
+                compilerResult.setCompilerMessages(compilerMessage);
+                compilerResult.setSuccess(false);
+            }
+        }
+
+        return compilerResult;
+    }
+
+    private List<CompilerMessage> compileFile(String sourceFile){
+        ExecutableModellingResult executableModellingResult ;
+        List<CompilerMessage> compilerMessage = new ArrayList();
+
+
+        try {
+            SlangSource slangSource = SlangSource.fromFile(new File(sourceFile));
+            executableModellingResult = slangCompiler.preCompileSource(slangSource);
+            if(executableModellingResult.getErrors() != null && !executableModellingResult.getErrors().isEmpty()){
+                for (RuntimeException runtimeException : executableModellingResult.getErrors()) {
+                    compilerMessage.add(new CompilerMessage(sourceFile + ": " + runtimeException.getMessage(), CompilerMessage.Kind.ERROR)) ;
+                }
+            }
+        } catch (Exception e) {
+            compilerMessage.add(new CompilerMessage(sourceFile + ": " + e.getMessage(), CompilerMessage.Kind.ERROR)) ;
+        }
+
+        return compilerMessage;
+    }
+
+    public String[] createCommandLine(CompilerConfiguration config) throws CompilerException {
+        return null ;
+    }
+
+    protected static String[] getSourceFiles( CompilerConfiguration config ){
+        Set sources = new HashSet();
+
+        for(Iterator it = config.getSourceLocations().iterator(); it.hasNext();){
+            String sourceLocation = (String) it.next();
+            sources.addAll(getSourceFilesForSourceRoot(config, sourceLocation ));
+        }
+
+        String[] result;
+        if ( sources.isEmpty() ){
+            result = new String[0];
+        } else {
+            result = (String[]) sources.toArray( new String[sources.size()] );
+        }
+
+        return result;
+    }
+
+    // we need to override this as it is hard coded java file extensions
+    protected static Set getSourceFilesForSourceRoot(CompilerConfiguration config, String sourceLocation) {
```

Comment:
maybe use collections with types instead of raw format -> Set<String> in these methods